### PR TITLE
qemu_io_blkdebug: replace perror with  os.strerror

### PR DESCRIPTION
**Pull request by pingl2016**
As command perror is not installed in some hosts, replace
perror with os.strerror

Signed-off-by: Ping Li <pingl@redhat.com>

ID: 1409205

### CODE REVIEW

File: qemu/tests/qemu_io_blkdebug.py [hunk 1]
```diff
@@ -110,15 +109,12 @@ def run(test, params, env):
...
             image_snapshot.remove()
 
-        error.context("Get error message from command perror", logging.info)
-        perror_cmd = "perror %s" % errn
-        std_msg = utils.system_output(perror_cmd)
-        std_msg = re.findall(re_std_msg, std_msg)
-        if std_msg:
-            std_msg = std_msg[0]
-        else:
+        error.context("Get error message", logging.info)
```
Comment (xutian) [inline qemu/tests/qemu_io_blkdebug.py]: Hi @pingl2016 
how about update your change like

```
from avocado.utils.path as utils_path
error.context("Get error message", logging.info)
try:
      err_cmd = utils_path.find_command('perror')
       out = utils.system_output('perror %s' % errn)
       ....
except utils_path.CmdNotFoundError:
       try:
            os.strerror(int(errn))
       except ValueError:
            logging.warn("Not error message found via : os.strerr(%s)" % errn)
```
above code can keep original logical.

Comment (pingl2016) [inline qemu/tests/qemu_io_blkdebug.py]: Both perror and os.strerror() can return the error message for the giving error code. it is better to use strerror() directly, then we don't need to take care whether command perror is installed on the host.

Comment (xutian) [inline qemu/tests/qemu_io_blkdebug.py]: OK.

**Comment by xutian** [comment]
Acked

**Comment by CongLi** [comment]
In commit log, typo: iqemu_io_blkdebug

**Comment by pingl2016** [comment]
@CongLi updated, thanks

File: qemu/tests/qemu_io_blkdebug.py [hunk 2]
```diff
@@ -110,15 +109,12 @@ def run(test, params, env):
...
+        try:
+            std_msg = os.strerror(int(errn))
+        except ValueError:
             std_msg = ""
-            logging.warning("Can not find error message from perror")
+            logging.warning("Can not find error message")
```
Comment (suqinhuang) [inline qemu/tests/qemu_io_blkdebug.py]: is it acceptable if not value? I found only warning here

File: qemu/tests/qemu_io_blkdebug.py [hunk 3]
```diff
@@ -110,26 +109,19 @@ def run(test, params, env):
...
-                fail_log += "qemu-io reports: '%s'," % output
-                fail_log += "perror reports: '%s'" % std_msg
-                raise error.TestFail(fail_log)
+        if std_msg in output:
+            logging.info("Error message is correct in qemu-io")
         else:
-            logging.warning("Can not find error message from perror."
-                            " The output from qemu-io is %s" % output)
+            fail_log = "The error message is mismatch:"
+            fail_log += "qemu-io reports: '%s'," % output
```
Comment (suqinhuang) [inline qemu/tests/qemu_io_blkdebug.py]: how about add "\n" to these lines

**Comment by pingl2016** [comment]
@suqinhuang updated, thanks for you comments.
'01-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_update_part' Finshed 2017-03-20 05:46:19 2017-03-20 05:46:26 PASS 7.326934
'02-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.cluster_alloc' Finshed 2017-03-20 05:46:27 2017-03-20 05:46:30 PASS 2.994056
'03-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.vmstate_load' Finshed 2017-03-20 05:46:30 2017-03-20 05:46:33 PASS 3.736743
'04-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l1_grow_alloc_table' Finshed 2017-03-20 05:46:33 2017-03-20 05:46:36 PASS 2.978576
'05-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l1_grow_activate_table' Finshed 2017-03-20 05:46:36 2017-03-20 05:46:43 PASS 6.825036
'06-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_alloc_write' Finshed 2017-03-20 05:46:43 2017-03-20 05:46:46 PASS 3.003394
'07-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.write_aio' Finshed 2017-03-20 05:46:46 2017-03-20 05:46:49 PASS 2.6605
'08-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.vmstate_save' Finshed 2017-03-20 05:46:49 2017-03-20 05:46:52 PASS 2.470919
'09-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.reftable_grow' Finshed 2017-03-20 05:46:52 2017-03-20 05:46:55 PASS 3.111345
'10-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_load' Finshed 2017-03-20 05:46:55 2017-03-20 05:46:58 PASS 3.009624
'11-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_alloc_write_table' Finshed 2017-03-20 05:46:58 2017-03-20 05:47:01 PASS 3.120482
'12-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_alloc_switch_table' Finshed 2017-03-20 05:47:01 2017-03-20 05:47:04 PASS 3.196179
'13-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_alloc_hookup' Finshed 2017-03-20 05:47:04 2017-03-20 05:47:08 PASS 3.184655
'14-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.read_aio' Finshed 2017-03-20 05:47:08 2017-03-20 05:49:26 PASS 137.949074
'15-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l2_update' Finshed 2017-03-20 05:49:26 2017-03-20 05:49:33 PASS 7.194006
'16-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l2_alloc_write' Finshed 2017-03-20 05:49:33 2017-03-20 05:49:40 PASS 6.860693
'17-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l1_update' Finshed 2017-03-20 05:49:40 2017-03-20 05:49:47 PASS 6.774072
'18-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l1_grow_write_table' Finshed 2017-03-20 05:49:47 2017-03-20 05:49:53 PASS 6.589256
'19-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.refblock_alloc' Finshed 2017-03-20 05:49:54 2017-03-20 05:49:57 PASS 2.976844
'20-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l2_load' Finshed 2017-03-20 05:49:57 2017-03-20 06:28:24 PASS 2307.262497
'21-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.l2_alloc_cow_read' Finshed 2017-03-20 06:28:24 2017-03-20 07:15:31 PASS 2827.296932
'22-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.qemu_io_blkdebug.cluster_free' Finshed 2017-03-20 07:15:31 2017-03-20 07:15:35 PASS 3.58509

**Comment by suqinhuang** [comment]
LGTM acked-by: Suqin Huang

**Pull request merged**